lib/elixir/lib/keyword.ex

```diff
@@ -867,6 +867,29 @@ defmodule Keyword do
     raise KeyError, key: key, term: original
   end
 
+  @doc """
+  Replaces the value under `key` using the given function only if
+  `key` already exists in `keywords`.
+
+  If `key` does not exist, the original keyword list is returned unchanged.
+
+  ## Examples
+      iex> Keyword.replace_lazy([{:a, 1}, {:b, 2}], :a, fn v -> v * 4 end)
+      [{:a, 4}, {:b, 2}]
+
+      iex> Keyword.replace_lazy([{:a, 1}, {:b, 2}], :c, fn v -> v * 4 end)
+      [{:a, 1}, {:b, 2}]
+  """
+  @doc since: "1.14.0"
+  @spec replace_lazy(t, key, (existing_value :: value -> new_value :: value)) :: t
+  def replace_lazy(keywords, key, fun) when is_list(keywords) and is_atom(key) and is_function(fun, 1) do
+    if Keyword.has_key?(keywords, key) do
+      Keyword.replace(keywords, key, fun.(keywords[key]))
+    else
+      keywords
+    end
+  end
```

Comment:
This traverses the keyword twice. You can look at the implementation of `Keyword.replace/3` for how to implement it using recursion that only traverses the keyword once.